### ISSUE TITLE
Replacement of binary fails if existing version has newlines

### ISSUE THREAD

**Issue by mholt** [caddyserver/getcaddy.com#28]
Some binaries of Caddy are built on a "dirty" working copy, in other words, changes made that were not committed. This is common for development machines that work on Caddy.

In those cases, the output of `caddy -version` can look like this:

```
Caddy 0.10.3 (+d3fc9f7 Tue Jun 27 18:07:28 UTC 2017)
 2 files changed, 11 insertions(+), 2 deletions(-)
caddy/caddymain/run.go
caddyhttp/proxy/reverseproxy.go
```

And this apparently breaks the install script [here](https://github.com/caddyserver/getcaddy.com/blob/4c056c98014eb0f414bc2d6ad4b7c9712ecc43be/index.txt#L158) because the $caddy_cur_ver variable is build with [this cut command](https://github.com/caddyserver/getcaddy.com/blob/4c056c98014eb0f414bc2d6ad4b7c9712ecc43be/index.txt#L151) and I don't think it considers the case where the version output looks like the above.

This isn't a huge priority and I've only seen it once myself, but I'm documenting this in case someone with bash chops who cares has a few minutes to see about remedying this. Or if not, that's fine... at least we know it's a potential issue now. Or I can get better at bash. :smile:

**Comment by pepa65** [comment]
So how would you like the backup to be called? Not caddy_0.10.3 apparently; would you like to see the whole string in the name, or is just the commit id sufficient?

(I wouldn't have thought it would be an issue; you just backup the last working binary in case of the upgraded version not working well.)

**Comment by mholt** [comment]
@pepa65 Good question, probably the commit ID, but I don't know the complexity of doing that!

Edit: Would it be easier to just call it caddy-old or something? The version can always be obtained with `caddy-old -version` if needed... :P

**Comment by pepa65** [comment]
It seems like how it is now (caddy_VERSION) is better than caddy_old, in case people would like to keep the latest commit/version of the main version number. If called caddy_old, only one backup will be kept. It's up to people's needs I guess. I can easily get the commit ID and use caddy_VERSION_COMMITID or somesuch.

**Comment by mholt** [comment]
It seems unlikely that people would need more than one backup, honestly. It would also be simpler. I am okay with just one. I can make that change, since we don't need the version variable, I think I can figure it out. :D

**Comment by pepa65** [comment]
I think it would become like this:
```bash
  # Back up existing caddy, if any
  if [[ -x $caddy_bin ]]; then
    caddy_path="$(type -p "$caddy_bin")"
    echo "Backing up $caddy_path to ${caddy_path}_old"
    echo "(Password may be required.)"
    $sudo_cmd mv "$caddy_path" "${caddy_path}_old"
  fi
```

**Comment by mholt** [comment]
@pepa65 That looks good, do you want to make a PR to get the credit? Otherwise I can put that change (or something similar) in the file.

**Comment by mholt** [comment]
Closed by #31
